src/mlpack/core/optimizers/sdp/lrsdp_function.hpp

```diff
@@ -57,110 +73,86 @@ class LRSDPFunction
                           arma::mat& gradient) const;
 
   //! Get the number of sparse constraints in the LRSDP.
-  size_t NumSparseConstraints() const { return sparseB.n_elem; }
+  size_t NumSparseConstraints() const { return sdp.NumSparseConstraints(); }
```

Comment:
Should we just provide an interface to the SDP itself with `SDP()` and then let the user get the constraints/etc. themselves?